HeterogeneousCore/CUDAUtilities/interface/prefixScan.h

```diff
@@ -149,21 +150,23 @@ namespace cms {
       if (!isLastBlockDone)
         return;
 
+      assert(int(gridDim.x)==*pc);
+
       // good each block has done its work and now we are left in last block
 
       // let's get the partial sums from each block
-      __shared__ T psum[1024];
+      extern __shared__ T psum[];
```

Comment:
the real reason to use dynamic shared is that cuda will reserve this shared space for all blocks even if only one will actually use it. Given that in most of the cases the number of blocks is much, much smaller (<100) this should put less pressure to resources (and therefore schedule more blocks in parallel)

Reply:
> the real reason to use dynamic shared is that cuda will reserve this shared space for all blocks even if only one will actually use it.

Do you mean that we can use a different size for different invocations of the kernel, or a different size for different blocks in the same invocation ?
I don't think the latter is supported.

Reply:
the first: that we can use a different size for different invocations of the kernel.